### ISSUE TITLE
New rule: `no-useless-lookaround-assertions`

### ISSUE THREAD

**Issue by ota-meshi** [ota-meshi/eslint-plugin-regexp#481]
**Motivation**

We can define unnecessary lookaround assertions within lookaround assertions, so I would like a rule to report this and fix it automatically.

```js
'JavaScript'.replace(/Java(?=Scrip(?=t))/u, 'Type');
'JavaScript'.replace(/(?<=(?<=J)ava)Script/u, '');
```

**Description**

New rule report unnecessary lookaround assertions within lookaround assertions.

**Examples**

```js
/* ✗ BAD */
'JavaScript'.replace(/Java(?=Scrip(?=t))/u, 'Type');
'JavaScript'.replace(/(?<=(?<=J)ava)Script/u, '');

/* ✓ GOOD */
'JavaScript'.replace(/Java(?=Script)/u, 'Type');
'JavaScript'.replace(/(?<=Java)Script/u, '');
```
